### PR TITLE
Fix compilation of nested `super(...)` calls

### LINKED ISSUE #14652: [Bug]: More Than One Super Call in the Construction is Incorrectly Allowed
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

@babel/cli

### Input code

```js
class envs {
    constructor(param, obj) {
        console.log(param)
    }
}
function obj2() {
    return 2;
}
function battlePlayer() {
    return 1;
}
class doSomethingMore extends envs {
    constructor() {
        super(super(), obj2());
        // super(obj2(), battlePlayer());
    }
}
b = new doSomethingMore()
```
[It's the transpiled one](https://babel.dev/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=usage&corejs=3.21&spec=true&loose=true&code_lz=FAYwNghgzlAECmA7AbnA3sWXYgPaKgBcAnAVxEN2IAoAHCYiAWwBpZcAjAKwEpYNsgnPii4w8AHRhcAczoNmPTNgC-wYGoBmpRBQCW-dtwBM1PgOzF4hUsUSxjAbg3BtuwgfscIhQuIAKkACe8DTmylhWNnawAIzOauDQcAAmuADKuEzWABZ6iDIAslTwCAAehEgpcEio_BHCBCTklGH1QthQpLSh1F09YWycXKY8PM4dWAD0U7D9vcOjbN6-AcG9YxOq6mocsAC8sIjwAO6waZnZhHkFxVZmwEA&debug=false&forceAllTransforms=true&shippedProposals=true&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=unambiguous&lineWrap=false&presets=env&prettier=false&targets=Node-10.13&version=7.18.4&externalPlugins=&assumptions=%7B%22superIsCallableConstructor%22%3Atrue%2C%22skipForOfIteratorClosing%22%3Atrue%2C%22setSpreadProperties%22%3Atrue%2C%22setPublicClassFields%22%3Atrue%2C%22setComputedProperties%22%3Atrue%2C%22setClassMethods%22%3Atrue%2C%22pureGetters%22%3Atrue%2C%22privateFieldsAsProperties%22%3Atrue%2C%22objectRestNoSymbols%22%3Atrue%2C%22noNewArrows%22%3Atrue%2C%22noDocumentAll%22%3Atrue%2C%22noClassCalls%22%3Atrue%2C%22mutableTemplateObject%22%3Atrue%2C%22ignoreToPrimitiveHint%22%3Atrue%2C%22iterableIsArray%22%3Atrue%2C%22ignoreFunctionLength%22%3Atrue%2C%22enumerableModuleMeta%22%3Atrue%2C%22constantSuper%22%3Atrue%2C%22constantReexports%22%3Atrue%2C%22arrayLikeIsIterable%22%3Atrue%7D)
[It is the report error one](https://babel.dev/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=usage&corejs=3.21&spec=true&loose=true&code_lz=FAYwNghgzlAECmA7AbnA3sWXYgPaKgBcAnAVxEN2IAoAHCYiAWwBpZcAjAKwEpYNsgnPii4w8AHRhcAczoNmPTNgC-wYGoBmpRBQCW-dtwBM1PgOzF4hUsUSxjAbg3BtuwgfscIhQuIAKkACe8DTmylhWNnawAIzOauDQcAAmuADKuEzWABZ6iDIAslTwCAAehEgpcEio_BHCBCTklGH1QthQpLSh1F09YWycXKY8PM4dWP29w6Ns3r4Bwb1jE6rqahywALywiPAA7rBpmdmEeQXFVmbAQA&debug=false&forceAllTransforms=true&shippedProposals=true&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=unambiguous&lineWrap=false&presets=env&prettier=false&targets=Node-10.13&version=7.18.4&externalPlugins=&assumptions=%7B%22superIsCallableConstructor%22%3Atrue%2C%22skipForOfIteratorClosing%22%3Atrue%2C%22setSpreadProperties%22%3Atrue%2C%22setPublicClassFields%22%3Atrue%2C%22setComputedProperties%22%3Atrue%2C%22setClassMethods%22%3Atrue%2C%22pureGetters%22%3Atrue%2C%22privateFieldsAsProperties%22%3Atrue%2C%22objectRestNoSymbols%22%3Atrue%2C%22noNewArrows%22%3Atrue%2C%22noDocumentAll%22%3Atrue%2C%22noClassCalls%22%3Atrue%2C%22mutableTemplateObject%22%3Atrue%2C%22ignoreToPrimitiveHint%22%3Atrue%2C%22iterableIsArray%22%3Atrue%2C%22ignoreFunctionLength%22%3Atrue%2C%22enumerableModuleMeta%22%3Atrue%2C%22constantSuper%22%3Atrue%2C%22constantReexports%22%3Atrue%2C%22arrayLikeIsIterable%22%3Atrue%7D)

### Configuration file name

babel.config.json

### Configuration

```json
{
  "sourceType": "unambiguous",
  "presets": ["@babel/preset-env"],
  "plugins": [
    "@babel/plugin-transform-runtime"
  ]
}
```

### Current and expected behavior

In the poc.js, super() in the constructor is called more than once, it is the wrong syntax, but it transpiled by babel.
However, if I add code to constructor:
```js
super(obj2(), battlePlayer());
```

it can report 
```shell
Container is falsy
```

### Environment

- babel 
- - ├── @babel/cli@7.17.0=6
- - ├── @babel/generator@7.17.3
- - ├── @babel/plugin-transform-runtime@7.17.0
- - ├── @babel/preset-env@7.16.11
- Node: [v17.3.0]
- npm version [8.5.2]
- os [ubuntu 20.04]

### Possible solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dbacarel**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |  Fixes #14652 
| Patch: Bug Fix?          | transform-classes
| Major: Breaking Change?  | 
| Minor: New Feature?      | 
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

The bug along with the fix are described in details in the issue. No need to duplicate I think .

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14709"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
I edited your description adding "Fixes" before the issue number. By doing so, it will be automatically closed when this PR is merged :slightly_smiling_face:

**Comment by nicolo-ribaudo** [comment]
The failing tests is a network error, I'll restart it.

File: packages/babel-plugin-transform-classes/src/transformClass.ts [hunk 1]
```diff
@@ -399,22 +399,22 @@ export default function transformClass(
       );
     }
 
-    const bareSupers = new Set<NodePath<t.CallExpression>>();
+    const bareSupers = new Array<NodePath<t.CallExpression>>();
```
Comment (JLHwung) [inline packages/babel-plugin-transform-classes/src/transformClass.ts:402]: Nit:
```suggestion
    const bareSupers: NodePath<t.CallExpression>[] = [];
```
will be shorter.

**Review by JLHwung — APPROVED** [review]

**Review by JLHwung — APPROVED** [review]
Thanks.

**Review by liuxingbaoyu — APPROVED** [review]

File: packages/babel-plugin-transform-classes/src/transformClass.ts (hunk 1 above)
Comment (dbacarel) [reply packages/babel-plugin-transform-classes/src/transformClass.ts]: 👌 done

**Pull request merged**